packages/wdio-browserstack-service/src/util.ts

```diff
@@ -526,7 +589,67 @@ export const getA11yResults = async (browser: WebdriverIO.Browser, isBrowserStac
     }
 }
 
-export const getA11yResultsSummary = async (browser: WebdriverIO.Browser, isBrowserStackSession?: boolean, isAccessibility?: boolean | string) : Promise<{ [key: string]: any; }> => {
+export const getAppA11yResults = async (isAppAutomate: boolean, browser: WebdriverIO.Browser, isBrowserStackSession?: boolean, isAccessibility?: boolean | string, sessionId?: string | null) : Promise<Array<{ [key: string]: any; }>> => {
+    if (!isBrowserStackSession) {
+        return [] // since we are running only on Automate as of now
+    }
+
+    if (!isAppAccessibilityAutomationSession(isAccessibility, isAppAutomate)) {
+        BStackLogger.warn('Not an Accessibility Automation session, cannot retrieve Accessibility results summary.')
+        return []
+    }
+
+    try {
+        BStackLogger.debug('Performing scan before getting results summary')
+        await performA11yScan(isAppAutomate, browser, isBrowserStackSession, isAccessibility)
+        const apiUrl = `${APP_ALLY_ENDPOINT}/${APP_ALLY_ISSUES_ENDPOINT}`
+        const upperTimeLimit = Date.now() + 360000 // 30 seconds
+        const params = { test_run_uuid: process.env.TEST_ANALYTICS_ID, session_id: sessionId, timestamp: Date.now() } // Query params to pass
+        const header = { Authorization: `Bearer ${process.env.BSTACK_A11Y_JWT}` }
+        console.log(`params ${JSON.stringify(params)}`)
+        console.log(`header ${JSON.stringify(header)}`)
+        const apiRespone = await pollApi(apiUrl, params, header, upperTimeLimit)
```

Comment:
where are we using polling Timeout? Also, logic seems different as compared to TestNG in terms of buffer seconds we are taking.

Reply:
What logic difference are there?

Reply:
I thought about this 360000 :) But this seems like a miss

Reply:
Let's keep it as 30 secs

Reply:
Changed